### PR TITLE
Update KU and EKU requirements

### LINKED ISSUE #130: id-kp-codeSigning EKU restriction for signing certificate.
Discuss if we need to enforce id-kp-codeSigning EKU restriction for signing certificate.

----

Recommend removing this requirement, as not all signed artifacts will be "code".

_Originally posted by @briankr-ms in https://github.com/notaryproject/notaryproject/pull/122#discussion_r798210495_

### PR DESCRIPTION
**Pull request by priteshbandi**
Updates KU and EKU requirements as per [certificate-requirements](https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#certificate-requirements) specification

- closes https://github.com/notaryproject/notaryproject/issues/130

### CODE REVIEW

File: x509/cert_validations.go [hunk 1]
```diff
@@ -10,11 +10,22 @@ import (
 	"time"
 )
 
+var kuLeafCertBlocked = x509.KeyUsageContentCommitment |
```
Comment (patrickzheng200) [inline x509/cert_validations.go:13]: One minor question here, the [certificate-requirements](https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#certificate-requirements) does not mention KeyUsageContentCommitment must be blocked for leaf certificates. Should we update the spec?

File: x509/cert_validations.go [hunk 2]
```diff
@@ -168,8 +179,15 @@ func validateExtendedKeyUsage(cert *x509.Certificate, expectedEku x509.ExtKeyUsa
...
 	}
 
-	excludedEkus := []x509.ExtKeyUsage{x509.ExtKeyUsageAny, x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageEmailProtection}
-	if expectedEku == x509.ExtKeyUsageCodeSigning {
+	excludedEkus := []x509.ExtKeyUsage{
+		x509.ExtKeyUsageAny,
+		x509.ExtKeyUsageServerAuth,
+		x509.ExtKeyUsageClientAuth,
+		x509.ExtKeyUsageEmailProtection,
+		x509.ExtKeyUsageOCSPSigning,
```
Comment (patrickzheng200) [inline x509/cert_validations.go:187]: same here, the spec didn't mention x509.ExtKeyUsageOCSPSigning

File: x509/cert_validations.go (hunk 1 above)
Comment (priteshbandi) [reply x509/cert_validations.go:13]: Yep, will add it to cert requirement.

**Review by patrickzheng200 — APPROVED** [review]
LGTM

**Review by jondonas — APPROVED** [review]

**Pull request merged**